### PR TITLE
[WIP] Added unit tests for one-shot object detection

### LINKED ISSUE #1788: Unit tests for One Shot Object Detection
Assign to Project 5.6 Sprint 2 when board becomes available.

### PR DESCRIPTION
**Pull request by srikris**
Fixes #1788

### CODE REVIEW

**Comment by srikris** [comment]
I'll merge this once all the open PRs for one shot are merged.

**Comment by TobyRoseman** [comment]
How long does this test take to run?

File: src/unity/python/turicreate/test/test_one_shot_object_detector.py [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+
+        self.get_ans = {
+           'classes': lambda x: x == sorted(_CLASSES),
+           'target': lambda x: x == self.feature,
+           'num_starter_images': lambda x: x > 0,
+           'num_classes': lambda x: x == len(_CLASSES),
+        }
+        self.fields_ans = self.get_ans.keys()
+
+    @pytest.mark.xfail(raises = _ToolkitError)
```
Comment (nickjong) [inline src/unity/python/turicreate/test/test_one_shot_object_detector.py]: Use `self.assertRaises`, as in our other tests. `xfail` is not for tests that are "designed" to fail. See also https://docs.pytest.org/en/latest/assert.html#assertions-about-expected-exceptions

**Comment by shreyajain17** [comment]
Moving to #2021

**Pull request closed without merging**